### ISSUE TITLE
2.12 nightly not found if specified, even if found with `2.12.nightly`

### ISSUE THREAD

**Issue by SethTisue** [VirtusLab/scala-cli#996]
this works and retrieves what is, at the moment, the latest 2.12 nightly:

```
% scala-cli repl -S 2.12.nightly
Welcome to Scala 2.12.16-bin-5bd9520 (OpenJDK 64-Bit Server VM, Java 17.0.2).
Type in expressions for evaluation. Or try :help.

scala>
```

but if I ask for the same version by number it isn't found:

```
% scala-cli repl -S 2.12.16-bin-5bd9520
[error]  Cannot find a valid matching Scala version among 2.11.9, 2.11.9-bin-03d5f4e, 2.11.9-bin-3f437a2, 2.11.9-bin-8d16a0b, 2.11.9-bin-42f813e, 2.11.9-bin-99f41a1, 2.11.9-bin-363d9e5, 2.11.9-bin-828a892, 2.11.10, 2.11.10-bin-d9150aa, 2.11.10-bin-7fa3823, 2.11.10-bin-94dd1dc, 2.11.10-bin-6008e4b, 2.11.11-bin-a3987d8, 2.11.11-bin-7a5df57, 2.11.11-bin-8a413ba, 2.11.11-bin-5167b69, 2.11.12, 2.11.12-bin-b8155a5, 2.11.12-bin-ad19744, ...
```

I've omitted reams of output that follow — a very long list of, apparently, every Scala 2 nightly ever published.

I'm not sure, but It's possible that I tried to retrieve this nightly before it had actually been published, and then perhaps that failure got cached? If that's what happened here, I don't know what cache I would clear to make it be found again?

**Comment by lwronski** [comment]
Hi @SethTisue, thanks for reporting
I ran the following command and it works for me:
```
lwronski@VL-D-0317 ~ % scala-cli repl -S 2.12.16-bin-5bd9520
Welcome to Scala 2.12.16-bin-5bd9520 (OpenJDK 64-Bit Server VM, Java 17.0.2).
Type in expressions for evaluation. Or try :help.
```

Can you check again? 

ScalaCLI has a mechanism that checks if someone passed the correct Scala version and in the cache the are lists the available scala versions. It seems the list of scala versions is refreshed too infrequently.

**Comment by SethTisue** [comment]
It still didn't work.

But then I moved my `~/Library/Caches/Coursier` directory aside, and that made it start working.

**Comment by lwronski** [comment]
It should be fixed by #1006,
from the next version of scala-cli `0.1.6` the list of correct scala nightly versions should be refreshed each time, so this problem shouldn't happen again.

Before,  the list of correct nightly scala versions was refreshed each 24 hours, therefore you can not run the latest 2.12 nightly

**Comment by SethTisue** [comment]
awesome, thank you!
